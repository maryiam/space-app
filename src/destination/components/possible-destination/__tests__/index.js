import renderer from 'react-test-renderer';
import React from 'react';
import PossibleDestination from '../index';

describe('possible-destination/', () => {
  describe('#render', () => {
    it('renders correctly with default props', () => {
      const tree = renderer.create(<PossibleDestination />);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with provided destination data', () => {
      const tree = renderer.create(
        <PossibleDestination
          onChange={jest.fn}
          destination={{
            id: 10,
            thumb: '/assets/planets/thumb/yavin4@x2.png',
            name: 'destination 1',
          }}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('callback invocation', () => {
    it('invokes the function with the id passed as prop when on the click action', () => {
      const mockFn = jest.fn();
      const tree = renderer.create(
        <PossibleDestination
          onChange={mockFn}
          destination={{
            id: 10,
            thumb: '/assets/planets/thumb/yavin4@x2.png',
            name: 'destination 1',
          }}
        />
      );

      const rootInstance = tree.root;
      const div = rootInstance.findByType('div');

      div.props.onClick();

      expect(mockFn).toBeCalledWith(10);
    });
  });
});
