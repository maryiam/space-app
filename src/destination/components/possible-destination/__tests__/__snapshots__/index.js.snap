// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`possible-destination/ #render renders correctly with default props 1`] = `
<div
  className="sc-bdVaJa isJPnW"
  onClick={[Function]}
>
  <img
    className="sc-htpNat iVofcY"
    src={null}
  />
  <span
    className="sc-bwzfXH kwcotr"
  >
    unknown
  </span>
</div>
`;

exports[`possible-destination/ #render renders correctly with provided destination data 1`] = `
<div
  className="sc-bdVaJa isJPnW"
  onClick={[Function]}
>
  <img
    className="sc-htpNat iVofcY"
    src="/assets/planets/thumb/yavin4@x2.png"
  />
  <span
    className="sc-bwzfXH kwcotr"
  >
    destination 1
  </span>
</div>
`;
